test/v3/controllers/requestParsers/SubmitForeignPropertyBsasRequestParserSpec.scala

```diff
@@ -62,12 +64,27 @@ class SubmitForeignPropertyBsasRequestParserSpec extends UnitSpec {
           )
 
         parser.parseRequest(inputData) shouldBe
-          Right(SubmitForeignPropertyBsasRequestData(Nino(nino), calculationId, body))
+          Right(SubmitForeignPropertyBsasRequestData(Nino(nino), calculationId, None, body))
+      }
+
+      "valid request data is supplied with a taxYear" in new Test {
+        val inputData: SubmitForeignPropertyRawData = inputDataWith(Some("2023-24"))
+        MockValidator.validate(inputData).returns(Nil)
+
+        val body: SubmitForeignPropertyBsasRequestBody =
```

Comment:
Could we extract this `body` variable from each test so we don't have to repeat it?

Reply:
Will do.